src/Components/Lifecycle/Lifecycle.tsx

```diff
@@ -701,10 +702,38 @@ const LifecycleTab: React.FC<React.PropsWithChildren> = () => {
     </Bullseye>
   );
 
+  // Cannot use the ErrorState with customFooter since there are errors with it.
+  // This is basically the same as ErrorState only with custom action at the bottom.
+  const timeoutState = (
+    <Bullseye>
+      <EmptyState variant={EmptyStateVariant.sm}>
+        <EmptyStateHeader
+          // the color could be imported from @patternfly/react-tokens/dist/esm/global_danger_color_100,
+          // but the import doesn't work
+          icon={<EmptyStateIcon icon={ExclamationCircleIcon} color="var(--pf-v5-global--danger-color--100)" />}
+          titleText="Timeout reached when calculating response"
+          headingLevel="h2"
+        />
+        <EmptyStateBody>This is a known issue that we are working to resolve.</EmptyStateBody>
+        <EmptyStateFooter>
+          <EmptyStateActions>
+            <Button variant="primary" onClick={redirectToDashboard}>
+              Return to home page
+            </Button>
+          </EmptyStateActions>
+        </EmptyStateFooter>
+      </EmptyState>
+    </Bullseye>
+  );
+
   if (error) {
     if (String(error.message) === 'Error: Workspace filtering is not yet implemented') {
       // corner case with workspace filtering, we need different error message
       return lockedState;
+    } else if (String(error.message) === 'Error: 504 Gateway Time-out') {
```

Comment:
This was tested with some of the endpoints set to raise an exception 
```raise HTTPException(detail="504 Gateway Time-out", status_code=504)```
It should match with the return value from the 3Scale layer, which kills the connection. This is something what should **definitely** be validated - I'm not sure if it's 100 % correct and cannot validate it locally (I don't have the 3Scale :D).

In case this won't work (like the output of the 3Scale is a bit different), we could probably use this condition 
```} else if (String(error.message).includes('504 Gateway Time-out')) {``` (and same for the `Upcoming.tsx`)

@SerCantus

Reply:
I'm going to change it now, since it seems the response from 3scale is `<html><body><h1>504 Gateway Time-out</h1>\nThe server didn't respond in time.\n</body></html>\n`, so te http exception raised on backend should look like:
```
raise HTTPException(
        status_code=504,
        detail="<html><body><h1>504 Gateway Time-out</h1>\nThe server didn't respond in time.\n</body></html>"
    )
```

but still something what should be checked

Reply:
It was possible, not sure if we need that anymore (since the timeout issue is probably fixed), but I believe having the status code could be useful even in the future.

Reply:
It will always be more accurate to check the status code that matching on message text since the text could change at some point in the future but status codes are well established.